base/logging_config.go

```diff
@@ -25,48 +28,47 @@ type LoggingConfig struct {
 	DeprecatedDefaultLog *LogAppenderConfig `json:"default,omitempty"` // Deprecated "default" logging option.
 }
 
-func (c *LoggingConfig) Init(defaultLogFilePath string) (err error) {
+func (c *LoggingConfig) Init(defaultLogFilePath string) (deferredLogs []DeferredLog, err error) {
 	if c == nil {
-		return errors.New("nil LoggingConfig")
+		return deferredLogs, errors.New("nil LoggingConfig")
 	}
 
-	consoleLogger, err = NewConsoleLogger(&c.Console)
+	consoleLogger, deferredLogs, err = NewConsoleLogger(&c.Console)
 	if err != nil {
-		return err
+		return
 	}
 
 	// If there's nowhere to specified put log files, we'll log an error, but we'll continue anyway.
 	if !hasLogFilePath(&c.LogFilePath, defaultLogFilePath) {
-		Errorf(KeyAll, "No logFilePath configured, and --defaultLogFilePath flag is not set. Log files required for product support are not being generated.")
-		return nil
+		return deferredLogs, ErrUnsetLogFilePath
 	}
 
 	err = validateLogFilePath(&c.LogFilePath, defaultLogFilePath)
 	if err != nil {
-		return err
+		return
```

Comment:
I personally prefer "non-naked returns", even if they're redundant, although I realize this might be more idiomatic go.  Taking a step back, I feel like as a team it would be great to have an agreed "coding style guide" document we follow, a la https://github.com/golang/go/wiki/CodeReviewComments

Reply:
Agreed, I dislike naked returns, will restore this.